File: server-scala-play-server/app/api/ListingApi.scala
package api

import play.api.libs.json._
import model.BigDecimal
import model.ListingFullResponse
import model.ListingGroupResponse
import model.ListingResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait ListingApi {
  /**
    * Create Listing
    * Creates a listing.
    * @param accountId the user&#39;s account ID
    * @param name the name of the listing
    * @param filterIds comma separated list of filter IDs
    * @param description the description of the listing
    * @param start the start date of the listing
    * @param end the end date of the listing
    * @param locationName the name of the location the listing will be held at
    * @param locationDescription the description of the location the listing will be held at
    * @param isPrivate determines whether the listing is public or private
    * @param externalId external identifier used by a third party
    * @param externalId2 secondary external identifier used by a third party
    * @param externalGroupId external group identifier used by a third party
    * @param active Sets the active flag
    * @param metaData external custom client defined data
    */
  def createListing(version: BigDecimal, accountId: Long, name: String, filterIds: Option[String], description: Option[String], start: Option[Long], end: Option[Long], locationName: Option[String], locationDescription: Option[String], isPrivate: Option[Boolean], externalId: Option[String], externalId2: Option[String], externalGroupId: Option[String], active: Option[Boolean], metaData: Option[String]): ListingFullResponse

  /**
    * Delete Listing
    * Delete a listing.
    * @param accountId the id of the logged in user
    * @param listingId the id of the listing to delete
    */
  def deleteListing(version: BigDecimal, accountId: Long, listingId: Long): SirqulResponse

  /**
    * Get Listing
    * Get a listing by id.
    * @param listingId the id of the listing to get
    */
  def getListing(version: BigDecimal, listingId: Long): ListingFullResponse

  /**
    * Search Listings
    * Search for event listings from the start time to end time
    * @param accountId the account id of the user
    * @param keyword search the event name and description for this keyword
    * @param start the record to begin the return set on
    * @param limit the number of records to return
    * @param activeOnly whether to search on active listings only
    * @param latitude the current latitude of the user
    * @param longitude the current longitude of the user
    * @param startDate the start date to search from
    * @param endDate the end date to search to
    * @param categoryIds only return items of these categories
    * @param filterIds 
    * @param useListingOrderIds determines whether to use configured listing order ids
    * @param externalId external identifier used by a third party
    * @param externalId2 secondary external identifier used by a third party
    * @param externalGroupId external group identifier used by a third party
    */
  def searchListing(version: BigDecimal, accountId: Option[Long], keyword: Option[String], start: Option[Int], limit: Option[Int], activeOnly: Option[Boolean], latitude: Option[Double], longitude: Option[Double], startDate: Option[Long], endDate: Option[Long], categoryIds: Option[String], filterIds: Option[String], useListingOrderIds: Option[Boolean], externalId: Option[String], externalId2: Option[String], externalGroupId: Option[String]): List[ListingResponse]

  /**
    * Summary Listing
    * Search for a list of summary listings from the start time up to 8 days out.
    * @param accountId the account id of the user
    * @param startDate the start date to search from
    * @param categoryIds the list of categories to search on
    * @param daysToInclude how far out to search, in days
    * @param useListingOrderIds determines whether to use configured listing order ids
    */
  def summaryListing(version: BigDecimal, accountId: Option[Long], startDate: Option[Long], categoryIds: Option[String], daysToInclude: Option[Int], useListingOrderIds: Option[Boolean]): List[ListingGroupResponse]

  /**
    * Update Listing
    * Updates a listing.
    * @param accountId the user&#39;s account ID
    * @param listingId the listing to update
    * @param filterIds comma separated list of filter IDs
    * @param name the name of the listing
    * @param description the description of the listing
    * @param start the start date of the listing
    * @param end the end date of the listing
    * @param locationName the name of the location the listing will be held at
    * @param locationDescription the description of the location the listing will be held at
    * @param isPrivate determines whether the listing is public or private
    * @param externalId external identifier used by a third party
    * @param externalId2 secondary external identifier used by a third party
    * @param externalGroupId external group identifier used by a third party
    * @param active Sets the active flag
    * @param metaData external custom client defined data
    */
  def updateListing(version: BigDecimal, accountId: Long, listingId: Long, filterIds: Option[String], name: Option[String], description: Option[String], start: Option[Long], end: Option[Long], locationName: Option[String], locationDescription: Option[String], isPrivate: Option[Boolean], externalId: Option[String], externalId2: Option[String], externalGroupId: Option[String], active: Option[Boolean], metaData: Option[String]): ListingFullResponse
}

File: server-ruby-sinatra/api/listing_api.rb
require 'json'


MyApp.add_route('POST', '/api/3.18/listing/create', {
  "resourcePath" => "/Listing",
  "summary" => "Create Listing",
  "nickname" => "create_listing",
  "responseClass" => "ListingFullResponse",
  "endpoint" => "/listing/create",
  "notes" => "Creates a listing.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the user&#39;s account ID",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "name",
      "description" => "the name of the listing",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "filter_ids",
      "description" => "comma separated list of filter IDs",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "the description of the listing",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "the start date of the listing",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "_end",
      "description" => "the end date of the listing",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "location_name",
      "description" => "the name of the location the listing will be held at",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "location_description",
      "description" => "the description of the location the listing will be held at",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "is_private",
      "description" => "determines whether the listing is public or private",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "false",
      "paramType" => "query",
    },
    {
      "name" => "external_id",
      "description" => "external identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_id2",
      "description" => "secondary external identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_group_id",
      "description" => "external group identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "active",
      "description" => "Sets the active flag",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "meta_data",
      "description" => "external custom client defined data",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/3.18/listing/delete', {
  "resourcePath" => "/Listing",
  "summary" => "Delete Listing",
  "nickname" => "delete_listing",
  "responseClass" => "SirqulResponse",
  "endpoint" => "/listing/delete",
  "notes" => "Delete a listing.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the logged in user",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "listing_id",
      "description" => "the id of the listing to delete",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/listing/get', {
  "resourcePath" => "/Listing",
  "summary" => "Get Listing",
  "nickname" => "get_listing",
  "responseClass" => "ListingFullResponse",
  "endpoint" => "/listing/get",
  "notes" => "Get a listing by id.",
  "parameters" => [
    {
      "name" => "listing_id",
      "description" => "the id of the listing to get",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/listing/search', {
  "resourcePath" => "/Listing",
  "summary" => "Search Listings",
  "nickname" => "search_listing",
  "responseClass" => "Array<ListingResponse>",
  "endpoint" => "/listing/search",
  "notes" => "Search for event listings from the start time to end time",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the account id of the user",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "keyword",
      "description" => "search the event name and description for this keyword",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "the record to begin the return set on",
      "dataType" => "Integer",
      "allowableValues" => "",
      "defaultValue" => "0",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "the number of records to return",
      "dataType" => "Integer",
      "allowableValues" => "",
      "defaultValue" => "20",
      "paramType" => "query",
    },
    {
      "name" => "active_only",
      "description" => "whether to search on active listings only",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "false",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "the current latitude of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "the current longitude of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start_date",
      "description" => "the start date to search from",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "end_date",
      "description" => "the end date to search to",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "category_ids",
      "description" => "only return items of these categories",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "filter_ids",
      "description" => "",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "use_listing_order_ids",
      "description" => "determines whether to use configured listing order ids",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "true",
      "paramType" => "query",
    },
    {
      "name" => "external_id",
      "description" => "external identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_id2",
      "description" => "secondary external identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_group_id",
      "description" => "external group identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/listing/summary', {
  "resourcePath" => "/Listing",
  "summary" => "Summary Listing",
  "nickname" => "summary_listing",
  "responseClass" => "Array<ListingGroupResponse>",
  "endpoint" => "/listing/summary",
  "notes" => "Search for a list of summary listings from the start time up to 8 days out.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the account id of the user",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start_date",
      "description" => "the start date to search from",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "category_ids",
      "description" => "the list of categories to search on",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "days_to_include",
      "description" => "how far out to search, in days",
      "dataType" => "Integer",
      "allowableValues" => "",
      "defaultValue" => "15",
      "paramType" => "query",
    },
    {
      "name" => "use_listing_order_ids",
      "description" => "determines whether to use configured listing order ids",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "defaultValue" => "true",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/3.18/listing/update', {
  "resourcePath" => "/Listing",
  "summary" => "Update Listing",
  "nickname" => "update_listing",
  "responseClass" => "ListingFullResponse",
  "endpoint" => "/listing/update",
  "notes" => "Updates a listing.",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the user&#39;s account ID",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "listing_id",
      "description" => "the listing to update",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "filter_ids",
      "description" => "comma separated list of filter IDs",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "name",
      "description" => "the name of the listing",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "the description of the listing",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "the start date of the listing",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "_end",
      "description" => "the end date of the listing",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "location_name",
      "description" => "the name of the location the listing will be held at",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "location_description",
      "description" => "the description of the location the listing will be held at",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "is_private",
      "description" => "determines whether the listing is public or private",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_id",
      "description" => "external identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_id2",
      "description" => "secondary external identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "external_group_id",
      "description" => "external group identifier used by a third party",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "active",
      "description" => "Sets the active flag",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "meta_data",
      "description" => "external custom client defined data",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

